website/pages/docs/nia/architecture.mdx

```diff
@@ -7,3 +7,45 @@ description: >-
 ---
 
 # Consul-Terraform-Sync Architecture
+
+Consul-Terraform-Sync is a service-oriented tool for managing
+network infrastructure near real-time. Consul-Terraform-Sync runs as a daemon
+and integrates the network topology maintained by your Consul cluster with your
+network infrastructure to dynamically secure and connect services.
+
+## 10,000 Foot View
+
+[![Consul-Terraform-Sync Architecture](/img/nia-highlevel-diagram.png)](/img/nia-highlevel-diagram.png)
+
+The diagram shows Consul-Terraform-Sync monitoring the Consul service catalog
+for updates and utilizing Terraform to update the state of the infrastructure.
+
+There are two principal aspects of Sync to know about corresponding to the
+lines to Consul and Terraform in the diagram above. The line to Consul
+represents the Watchers monitoring the state of Consul's service catalog (and
+possibly KV store) while the line to Terraform represents Tasks being run to
+update the infrastructure.
+
+## Watcher and Views
+
+Consul-Terraform-Sync monitors Consul for updates utilizing Consul's [Blocking
+Queries](/api-docs/features/blocking) whenever supported, falling back on
+polling when not. The watcher maintains a separate thread (known internally as
+a view) for each value monitored, running any Tasks that depend on that watched
+value whenever it's updated.
+
+## Tasks
+
+A task is the action triggered by the updated data monitored in Consul. It
+takes the that dynamic service data and translates it into a call to the
+infrastructure application to configure it with the updates. It uses a driver
+to push out these updates, the initial driver being a local Terraform run. An
+example of a task is to automate a firewall security policy rule with
+discovered IP addresses for a set of Consul services.
+
+## Drivers
+
+A driver encapsulates the resources required to communicate the updates to the
+network infrastructure. Terraform is the initial supported driver. It comes
+with a set of providers that enables supporting supporting a wide variety of
+infrastructure applications with little customization.
```

Comment:
> with little customization

Can you expand on what this means? The wording reads a bit confusing because Terraform and 
modules are very customizable.

Reply:
I'm trying to say that it doesn't take a lot of work to get a new backend working as long as a provider exists. That creating a module doesn't require a lot of work. But I wanted to express this without going into details, like mentioning modules, as the TF driver has its own section.

Maybe just "with little effort"?

Reply:
Ah I see now. "With little effort" is assuming about the user because it might actually be a lot of effort depending on background/experience.

If it might help with other ideas on how to phrase it, this is how I worded a similar message in the Network Driver page

> This expands the Consul ecosystem and taps into the rich features and community of Terraform and Terraform providers.

Reply:
Or "without a lot of work"?

Reply:
Chatted in slack and decided to just drop the "with little customzation" bit altogether.